### PR TITLE
Split TaskLoggerService out of SchedulerService

### PR DESCRIPTION
**Pull request by adamstruck**
Also made a distinction between a `Worker` and a `Node`. 

* `Worker` manages a single task execution.
* `Node` reports available resources to the Scheduler service and is assigned workers.

### CODE REVIEW

**Comment by adamstruck** [comment]
also updates grpc-go to 1.5.2

**Comment by adamstruck** [comment]
* Added detail view for `Node` in web dashboard. 
* Fixed bug in dashboard that caused unnecessary background polling processes.

**Comment by adamstruck** [comment]
This PR stems from the proposal below.  To move towards that direction, this PR separates out the components in funnel that deal with persistent workers.  The end goal here is to maintain funnel's basic scheduler as a compute backend rather than a core component. 

Proposal: Funnel should not manage persistent workers itself

https://docs.google.com/document/d/18mjeI77aA64bTBP8rDPC-iUcqeUdtCwawXrHfHZ81Iw/edit#heading=h.sij8lzz88pqs

**Comment by adamstruck** [comment]
includes fix for #206

**Comment by buchanae** [comment]
Could you an empty `.buildbot` file to make buildbot happy please?

Could you split the fix for #206 into a separate PR?

File: scheduler/util.go [hunk 1]
```diff
@@ -49,50 +100,53 @@ func ScheduleSingleTaskWorker(prefix string, c config.Worker, t *tes.Task) *Offe
...
+	nconf.ID = n.Id
+	nconf.Timeout = 5 * time.Second
+	nconf.Resources.Cpus = n.Resources.Cpus
+	nconf.Resources.RamGb = n.Resources.RamGb
+	nconf.Resources.DiskGb = n.Resources.DiskGb
 
-	confPath := path.Join(workdir, "worker.conf.yml")
-	wc.ToYamlFile(confPath)
+	c := conf
+	c.Scheduler.Node = nconf
```
Comment (buchanae) [inline scheduler/util.go]: If you want, you could make nconf a pointer, remove the copy of conf (to `c`) and write `conf` to the file.

Comment (buchanae) [reply scheduler/util.go]: Also, why does the node now need the full stack config?

File: server/scheduler_boltdb.go [hunk 2]
```diff
@@ -0,0 +1,291 @@
+package server
```
Comment (buchanae) [inline server/scheduler_boltdb.go]: I can't tell what changed here because the file was renamed. Can you change the name back?

File: server/server.go [hunk 3]
```diff
@@ -19,31 +20,33 @@ var log = logger.Sub("server")
...
+	SchedulerServiceServer  pbs.SchedulerServiceServer
+	Handler                 http.Handler
+	DisableHTTPCache        bool
+	DialOptions             []grpc.DialOption
 }
 
 // DefaultServer returns a new server instance.
-func DefaultServer(db Database, conf config.Config) *Server {
-	log.Debug("Server Config", "config.Config", conf)
+func DefaultServer(db Database, conf config.Server) *Server {
```
Comment (buchanae) [inline server/server.go]: Nice!

File: server/task_boltdb.go [hunk 4]
```diff
@@ -149,39 +106,27 @@ func (taskBolt *TaskBolt) CreateTask(ctx context.Context, task *tes.Task) (*tes.
...
 
 	taskID := util.GenTaskID()
+	idBytes := []byte(taskID)
 	log := log.WithFields("taskID", taskID)
 
-	jwt := getJWT(ctx)
-	log.Debug("JWT", "token", jwt)
-
-	ch := make(chan *tes.CreateTaskResponse, 1)
-	err := taskBolt.db.Update(func(tx *bolt.Tx) error {
```
Comment (buchanae) [inline server/task_boltdb.go]: I like the cleanups here, but there are too many unrelated changes in this PR.

File: webdash/app.js [hunk 5]
```diff
@@ -18,10 +18,10 @@ function listAllTasks($http, tasks, page) {
...
   }
 
-  return $http.get(url).then(function(result) {
-    Array.prototype.push.apply(tasks, result.data.tasks);
-    if (result.data.next_page_token) {
-      return listAllTasks($http, tasks, result.data.next_page_token);
+  return $http.get(url).then(function(response) {
+    Array.prototype.push.apply(tasks, response.data.tasks);
+    if (response.data.next_page_token) {
+      return listAllTasks($http, tasks, response.data.next_page_token);
```
Comment (buchanae) [inline webdash/app.js]: Ya, response is a better name.

File: webdash/app.js [hunk 6]
```diff
@@ -53,63 +58,145 @@ app.controller('TaskListController', function($scope, NgTableParams, $http, $int
...
+  refresh();
+  stop = $interval(refresh, 2000);
 
-  $scope.cancelTask = function(taskID) {
-    var url = "/v1/tasks/" + taskID + ":cancel";
-    $http.post(url);
-  }
+  $scope.$on('$destroy', function() {
+    $interval.cancel(stop);
+  });
```
Comment (buchanae) [inline webdash/app.js]: The changes above should be pulled into a separate PR.

File: webdash/app.js [hunk 7]
```diff
@@ -53,63 +58,145 @@ app.controller('TaskListController', function($scope, NgTableParams, $http, $int
...
+       }).
+       when('/nodes/:node_id', {
+         templateUrl: 'static/node.html'
+       }).
+       when('/notfound', {
+         templateUrl: 'static/error404.html'
+       }).
+       otherwise({
+         templateUrl: 'static/error404.html'
+       });
```
Comment (buchanae) [inline webdash/app.js]: Awesome, thank you. Can we pull this into a smaller fix as well please?

File: worker/step.go [hunk 8]
```diff
@@ -65,11 +65,11 @@ func (s *stepRunner) Run(ctx context.Context) error {
 	}
 }
 
-func (s *stepRunner) logTails() (*tailer, *tailer) {
-	stdout, _ := newTailer(s.Conf.LogTailSize, func(c string) {
+func (s *stepWorker) logTails() (*tailer, *tailer) {
+	stdout, _ := newTailer(s.Conf.BufferSize, func(c string) {
```
Comment (buchanae) [inline worker/step.go]: `BufferSize` doesn't describe what buffer it is referring to.

File: worker/worker.go [hunk 9]
```diff
@@ -2,163 +2,319 @@ package worker
...
-	conf      config.Worker
-	sched     scheduler.Client
-	log       logger.Logger
-	resources config.Resources
-	newRunner RunnerFactory
-	runners   *runSet
-	timeout   util.IdleTimeout
-	state     pbf.WorkerState
-}
+// Run runs the task runner.
```
Comment (buchanae) [inline worker/worker.go]: Needs update.

File: Makefile [hunk 10]
```diff
@@ -161,7 +166,7 @@ full: proto install prune_deps add_deps tidy lint test website webdash
 
 # Build the website
 website:
-	@find ./config -name '*.txt' -o -name '*.yaml' -exec cp {} website/static/funnel-config-examples/ \;
+	@find ./config -name '*.txt' -o -name '*.yaml' | xargs -I % cp % ./website/static/funnel-config-examples/
```
[the remark this answers is not in the document]
Comment (adamstruck) [reply Makefile]: this command was broken

File: server/scheduler_boltdb.go (hunk 2 above)
Comment (adamstruck) [reply server/scheduler_boltdb.go]: Sorry for the bad diff. You can see the changes by running: 

`git diff master:server/scheduler_service.go split-scheduler-service:server/scheduler_boltdb.go`

File: proto/scheduler/scheduler.proto [hunk 11]
```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
```
[the remark this answers is not in the document]
Comment (adamstruck) [reply proto/scheduler/scheduler.proto]: I think that a change like that belongs in a different PR.

File: server/task_boltdb.go [hunk 12]
```diff
@@ -40,17 +34,17 @@ var TasksLog = []byte("tasks-log")
 // ExecutorLogs maps (task ID + executor index) -> tes.ExecutorLog struct
 var ExecutorLogs = []byte("executor-logs")
 
-// Workers maps:
-// worker ID -> funnel.Worker struct
-var Workers = []byte("workers")
+// Nodes maps:
+// node ID -> pbs.Node struct
+var Nodes = []byte("pools")
```
Comment (buchanae) [inline server/task_boltdb.go]: Missed a "pools" here
